src/visualization/src/Visualizer.cpp

```diff
@@ -552,4 +584,49 @@ bool Visualizer::isWindowActive() const
 #endif
 }
 
+
+bool Visualizer::setColorPalette(const std::string &name)
+{
+#ifdef IDYNTREE_USES_IRRLICHT
+
+    if( !pimpl->m_isInitialized )
+    {
+        reportError("Visualizer",
+                    "setColorPalette",
+                    "Impossible to set the color palette. Please initialize the visualizer");
+        return false;
+    }
+
+    const auto colors = pimpl->m_palette.find(name);
+
+    if(colors == pimpl->m_palette.end())
+    {
+        std::string paletteName;
+        for (const auto& tmp: pimpl->m_palette)
+            paletteName += " " + tmp.first;
+
+        const std::string error = "The palette named " + name
+                                + " does not exist. The following palette are available"
+                                + paletteName + ".";
+
+        reportError("Visualizer","setColorPalette", error.c_str());
+        return false;
+    }
+
+    this->enviroment().setBackgroundColor(irrlicht2idyntree(colors->second.background));
+    this->enviroment().setFloorGridColor(irrlicht2idyntree(colors->second.gridColor));
+
+    // delete the frame in the origin and create a new one
+    pimpl->m_environment.m_rootFrameNode->drop();
+    pimpl->m_environment.m_rootFrameNode = addFrameAxes(pimpl->m_irrSmgr,
+                                                        pimpl->m_environment.m_envNode,
+                                                        pimpl->rootFrameArrowsDimension);
```

Comment:
You are not changing the color of the axes here, aren't you?

Reply:
ops